src/engine/Comparators.h

```diff
@@ -5,17 +5,19 @@
 
 #include <utility>
 #include <vector>
+#include "IdTable.h"
 
 using std::pair;
 using std::vector;
 
-template <typename E>
 class OBComp {
  public:
   OBComp(const vector<pair<size_t, bool>>& sortIndices)
       : _sortIndices(sortIndices) {}
 
-  bool operator()(const E& a, const E& b) const {
+  template <int I>
+  bool operator()(const typename IdTableImpl<I>::Row& a,
```

Comment:
That's what I tried out initially, but because the `Row` type in `IdTableStatic` is declared using a `using` statement the compiler had trouble inferring the value of `I` (which must be found using template argument deduction). Templating the entire comparator on the other hand leads to the problem, that the I needs to be determined within the `OrderBy` operation, which would require a set of 5 if clauses to determine the value of `I` for the comparator (as the `CALL_WITH_FIXED...` macros don't support this case).

Reply:
Interesting that `using` changes the behavior like that, do you think that is intended/correct or rather a compiler problem?

Reply:
I actually am not certain if this should work or not, but I think it might fall into the 'Aliases are never deduced' rule ([cppreference](https://en.cppreference.com/w/cpp/language/template_argument_deduction)).